src/Microsoft.AspNet.Server.Kestrel/Http/Frame.Generated.cs

```diff
@@ -0,0 +1,297 @@
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Server.Kestrel.Http 
+{
+    public partial class Frame
+    {
+        
+        private object _currentIHttpRequestFeature;
+        private object _currentIHttpResponseFeature;
+        private object _currentIHttpRequestIdentifierFeature;
+        private object _currentIHttpSendFileFeature;
+        private object _currentIServiceProvidersFeature;
+        private object _currentIHttpAuthenticationFeature;
+        private object _currentIHttpRequestLifetimeFeature;
+        private object _currentIQueryFeature;
+        private object _currentIFormFeature;
+        private object _currentIResponseCookiesFeature;
+        private object _currentIItemsFeature;
+        private object _currentIHttpConnectionFeature;
+        private object _currentITlsConnectionFeature;
+        private object _currentIHttpUpgradeFeature;
+        private object _currentIHttpWebSocketFeature;
+        private object _currentISessionFeature;
```

Comment:
Picked up in https://github.com/aspnet/KestrelHttpServer/pull/239 List should be faster for < 16 items on x86 and < 8 items on x64 as its a single memory access to get the array in L1 cache (64 bytes); then iterating the array is just processing L1 cache.

Store the 3 types  ```IHttpRequestFeature```, ```IHttpResponseFeature```, ```IHttpUpgradeFeature``` in their common ```this``` state as a single ```int``` and only use object cache for the always and common features; and list beyond that.